### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'add_job'

### DISCUSSION THREAD

**Discussion by tomhash2** [tuya/tuya-home-assistant#335] [Q&A]
Hello everyone,

I'm wondering if I am the only one seeing this error after upgrading to HA 2021.8.x:

Aug 10 22:10:35 HomeAssistant dd442c487859[18524]:   File "/config/custom_components/tuya_v2/__init__.py", line 163, in update_device
Aug 10 22:10:35 HomeAssistant dd442c487859[18524]:     ha_device.schedule_update_ha_state()
Aug 10 22:10:35 HomeAssistant dd442c487859[18524]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 604, in schedule_update_ha_state
Aug 10 22:10:35 HomeAssistant dd442c487859[18524]:     self.hass.add_job(self.async_update_ha_state(force_refresh))  # type: ignore
Aug 10 22:10:35 HomeAssistant dd442c487859[18524]: AttributeError: 'NoneType' object has no attribute 'add_job'

I have two bulbs and two sockets.

Anyone?

Tom.

**Comment by tsutsuku** [comment]
Please provide More logs, also welcome to create a issue.

**Reply by tomhash2** [reply]
Thank you for reply. When I'm trying to switch on socket, I get an error. Socket switches on but remains off in HA.

Aug 11 23:27:01 HomeAssistant dd442c487859[18524]: 2021-08-11 23:27:01 ERROR (Thread-5) [root] Uncaught thread exception
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]: Traceback (most recent call last):
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/threading.py", line 973, in _bootstrap_inner
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self.run()
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/threading.py", line 910, in run
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self._target(*self._args, **self._kwargs)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3452, in                                                         _thread_main
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self.loop_forever(retry_first_connection=True)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1779, in                                                         loop_forever
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     rc = self.loop(timeout, max_packets)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1181, in                                                         loop
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     rc = self.loop_read(max_packets)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1572, in                                                         loop_read
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     rc = self._packet_read()
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2310, in                                                         _packet_read
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     rc = self._packet_handle()
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2936, in                                                         _packet_handle
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     return self._handle_publish()
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3216, in                                                         _handle_publish
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self._handle_on_message(message)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3444, in                                                         _handle_on_message
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self.on_message(self, self._userdata, message)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/tuya_iot/openmq.py", line 157, in _o                                                        n_message
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     listener(msgDict)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 175, in on                                                        _message
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self._on_device_report(data["devId"], data["status"])
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 196, in _o                                                        n_device_report
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self.__update_device(device)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 183, in __                                                        update_device
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     listener.update_device(device)
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/config/custom_components/tuya_v2/__init__.py", line 164, in update_device
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     ha_device.schedule_update_ha_state()
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 604, in sched                                                        ule_update_ha_state
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]:     self.hass.add_job(self.async_update_ha_state(force_refresh))  # type: ignore
Aug 11 23:27:01 HomeAssistant dd442c487859[18524]: AttributeError: 'NoneType' object has no attribute 'add_job'
